src/renderer/components/cluster-manager/cluster-view.tsx

```diff
@@ -68,6 +68,11 @@ class NonInjectedClusterView extends React.Component<Dependencies> {
   bindEvents() {
     disposeOnUnmount(this, [
       reaction(() => this.clusterId, async (clusterId) => {
+        // TODO: replace with better handling
+        if (clusterId && !this.props.entityRegistry.getById(clusterId)) {
+          return this.props.navigateToCatalog(); // redirect to catalog when the clusterId does not correspond to an entity
```

Comment:
So when this bug happens user is redirected to the Catalog page, and then has to try to connect again, hoping it will work?

Reply:
So the bug is that the file for the cluster did exist so `main` has an instance of `Cluster` for it, but the file has been removed so when the `iframe` is created (which happens before the activation of the cluster succeeds) that file isn't there and thus throws an error in the constructor.

The other options for this is to change the constructor of the cluster so that the data from the file is not a requirement but I would want more tests before doing that.